xmind/tests/test_relationship_element.py

```diff
@@ -0,0 +1,324 @@
+
+        with patch.object(_relationship_element, 'getAttribute') as _get_attribute:
+            _get_attribute.return_value = 120
+            self.assertEqual(120, _relationship_element.getEnd2ID())
+
+        self._wb_mixin_element_init.assert_called_once_with(
+            _relationship_element, 12, None)
+        self._add_attribute.assert_called_once_with(
+            _relationship_element, ATTR_ID)
+        _get_attribute.assert_called_once_with("end2")
+
+    def test_setEnd2ID(self):
+        _relationship_element = RelationshipElement(12)
+        _set_attribute = patch.object(
+            _relationship_element, 'setAttribute').start()
+        _update_modified_time = patch.object(
+            _relationship_element, 'updateModifiedTime').start()
+
+        self.assertIsNone(_relationship_element.setEnd2ID(17))
+
+        _set_attribute.assert_called_once_with("end2", 17)
+        _update_modified_time.assert_called_once()
+        self._wb_mixin_element_init.assert_called_once_with(
+            _relationship_element, 12, None)
+        self._add_attribute.assert_called_once_with(
+            _relationship_element, ATTR_ID)
+
+    def test_getEnd1(self):
+        _relationship_element = RelationshipElement(12)
+
+        with patch.object(_relationship_element, '_find_end_point') as _find_end_point:
+            _find_end_point.return_value = 120
+            self.assertEqual(120, _relationship_element.getEnd1(111))
+
+        self._wb_mixin_element_init.assert_called_once_with(
+            _relationship_element, 12, None)
+        self._add_attribute.assert_called_once_with(
+            _relationship_element, ATTR_ID)
+        _find_end_point.assert_called_once_with(111)
+
+    def test_getTitle_title_is_none(self):
+        _relationship_element = RelationshipElement(12)
+        _title_element = patch('xmind.core.relationship.TitleElement').start()
+
+        with patch.object(_relationship_element, '_get_title') as _get_title:
+            _get_title.return_value = None
+            self.assertIsNone(_relationship_element.getTitle())
+
+        self._wb_mixin_element_init.assert_called_once_with(
+            _relationship_element, 12, None)
+        self._add_attribute.assert_called_once_with(
+            _relationship_element, ATTR_ID)
+        _get_title.assert_called_once()
+        _title_element.assert_not_called()
+
+    def test_getTitle(self):
+        _relationship_element = RelationshipElement(12)
+        _title_mock = Mock()
+        _title_mock.getTextContent.return_value = 'it works'
+        _title_element = patch('xmind.core.relationship.TitleElement').start()
+        _title_element.return_value = _title_mock
+        _get_owner_workbook = patch.object(
+            _relationship_element, 'getOwnerWorkbook').start()
+        _get_owner_workbook.return_value = 'somehow'
+
+        with patch.object(_relationship_element, '_get_title') as _get_title:
+            _get_title.return_value = 'None'

```

Comment:
'None' - as for me its a bad value, I have spent 5 min for invstigation before i understood that this is a string value - not None